andela/zinnia-ah-backend
# #165575539 Fix existing username validation error

#### What does this PR do?
Adds validation for existing username in user update 

#### Description of Task to be completed?
- [x] Validate input for existing username
- [x] Creates middleware for existing email and username for signup and updating profiles

#### How should this be manually tested?
- Clone the repo,
- Run ```npm install``` to install all dependencies,
- Create a .env file using the .env.example as a guide and enter your environment variables
- Ensure you have a PostgreSQL server running on your development environment
- Run `sequelize db:create`  
- Run `sequelize db:migrate` 
- Run `sequelize db:seed:all`   
- Run `npm run start:dev`
- Send a PUT to http://localhost:3000/api/v1/users/profile to Postman.

#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories?(if applicable)
[#165575539](https://www.pivotaltracker.com/story/show/165575539)
#### Screenshots
![image](https://user-images.githubusercontent.com/38259391/56687048-52809c80-66cd-11e9-95d4-306def21dcad.png)
